build-cmakelibs.sh

```diff
@@ -19,107 +22,110 @@ IOP_CMAKE_OPTIONS=("-DCMAKE_TOOLCHAIN_FILE=$PS2DEV/share/ps2dev_iop.cmake" "-DCM
 # We need to clone a fork, this is a PR opened for ading cmake support
 # https://github.com/xiph/theora/pull/14
-git clone --depth 1 -b feature/cmake https://github.com/mcmtroffaes/theora.git || { exit 1; }
+$FETCH feature/cmake https://github.com/mcmtroffaes/theora.git &
 
 # SDL requires to have gsKit
-git clone --depth 1 -b v1.3.7 https://github.com/ps2dev/gsKit || { exit 1; } 
+$FETCH v1.3.7 https://github.com/ps2dev/gsKit &
 
 # We need to clone the whole repo and point to the specific hash for now,
 # till a new version is released after this commit
-git clone https://github.com/libsdl-org/SDL.git || { exit 1; }
-(cd SDL && git checkout 10c14e78b650e626293aa18155efec54cdee7098 && cd -) || { exit 1; }
-git clone --depth 1 -b release-2.6.3 https://github.com/libsdl-org/SDL_mixer.git || { exit 1; }
-git clone --depth 1 -b release-2.6.3 https://github.com/libsdl-org/SDL_image.git || { exit 1; }
-git clone --depth 1 -b release-2.20.2 https://github.com/libsdl-org/SDL_ttf.git || { exit 1; }
+$FETCH 10c14e78b650e626293aa18155efec54cdee7098 https://github.com/libsdl-org/SDL.git &
+$FETCH release-2.6.3 https://github.com/libsdl-org/SDL_mixer.git &
+$FETCH release-2.6.3 https://github.com/libsdl-org/SDL_image.git &
+$FETCH release-2.20.2 https://github.com/libsdl-org/SDL_ttf.git &
 
 # We need to clone the whole repo and point to the specific hash for now,
 # till a new version is released after this commit
-git clone https://github.com/sahlberg/libsmb2.git || { exit 1; } 
-(cd libsmb2 && git checkout dccf1dbccd66a8c433e336de4951a249096e1c22 && cd -) || { exit 1; }
+$FETCH dccf1dbccd66a8c433e336de4951a249096e1c22 https://github.com/sahlberg/libsmb2.git &
 # We need to clone the whole repo and point to the specific hash for now,
 # till a new version is released after this commit
-git clone https://github.com/lsalzman/enet.git || { exit 1; }
-(cd enet && git checkout 7083138fd401faa391c4f829a86b50fdb9c5c727 && cd -) || { exit 1; }
+$FETCH 7083138fd401faa391c4f829a86b50fdb9c5c727 https://github.com/lsalzman/enet.git &
 
 # Use wget to download argtable2
-wget -c http://prdownloads.sourceforge.net/argtable/argtable2-13.tar.gz || { exit 1; }
-tar -xzf argtable2-13.tar.gz || { exit 1; }
-git clone --depth 1 -b v3.2.2.f25c624 https://github.com/argtable/argtable3.git || { exit 1; }
+wget -c http://prdownloads.sourceforge.net/argtable/argtable2-13.tar.gz &
+$FETCH v3.2.2.f25c624 https://github.com/argtable/argtable3.git &
+
+# wait for fetch jobs to finish
+wait
+
+# extract argtable2
+tar -xzf argtable2-13.tar.gz
+
+# NOTE: jsoncpp
+# "snprintf" not found in "std" namespace error may occur, so patch that out here.
+pushd jsoncpp
+sed -i -e 's/std::snprintf/snprintf/' include/json/config.h
+popd
 
 ##
 ## Build cmake projects
 ##
-PROC_NR=1 build zlib -DUNIX:BOOL=ON # Forcing to compile with -j1 because there is a race condition in zlib
+build zlib -DUNIX:BOOL=ON -DZLIB_BUILD_EXAMPLES=OFF
```

Comment:
I'm not sure about this change.....
Even in `PSP` they required to use that trick with the latest available `zlib` version.
https://github.com/pspdev/psp-packages/blob/master/zlib/PSPBUILD#L25

Reply:
Yeah I think I saw something like that with the zlib on my setup but I was not sure if it's intended or not and it somehow compiled for me.

**Edit:**
This is my output, with `-j 16` but I have active ccache.
```
[ 93%] Building C object CMakeFiles/zlibstatic.dir/zutil.c.o
[100%] Linking C static library libz.a
[100%] Linking C static library libz.a
[100%] Built target zlib
[100%] Built target zlibstatic
```

Without ccache and with examples turned on:
```
Scanning dependencies of target zlibstatic
Scanning dependencies of target zlib
[  5%] Building C object CMakeFiles/zlibstatic.dir/adler32.c.o
[  5%] Building C object CMakeFiles/zlib.dir/adler32.c.o
[ 19%] Building C object CMakeFiles/zlibstatic.dir/deflate.c.o
[ 19%] Building C object CMakeFiles/zlib.dir/crc32.c.o
[ 19%] Building C object CMakeFiles/zlib.dir/compress.c.o
[ 19%] Building C object CMakeFiles/zlibstatic.dir/crc32.c.o
[ 19%] Building C object CMakeFiles/zlibstatic.dir/compress.c.o
[ 44%] Building C object CMakeFiles/zlibstatic.dir/gzclose.c.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzwrite.c.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzread.c.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzlib.c.o
[ 44%] Building C object CMakeFiles/zlib.dir/deflate.c.o
[ 44%] Building C object CMakeFiles/zlibstatic.dir/gzwrite.c.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzclose.c.o
[ 44%] Building C object CMakeFiles/zlibstatic.dir/gzlib.c.o
[ 44%] Building C object CMakeFiles/zlibstatic.dir/gzread.c.o
[ 50%] Building C object CMakeFiles/zlibstatic.dir/inflate.c.o
[ 50%] Building C object CMakeFiles/zlib.dir/inflate.c.o
[ 55%] Building C object CMakeFiles/zlib.dir/infback.c.o
[ 55%] Building C object CMakeFiles/zlibstatic.dir/infback.c.o
[ 61%] Building C object CMakeFiles/zlibstatic.dir/inftrees.c.o
[ 61%] Building C object CMakeFiles/zlib.dir/inftrees.c.o
[ 66%] Building C object CMakeFiles/zlibstatic.dir/inffast.c.o
[ 66%] Building C object CMakeFiles/zlib.dir/inffast.c.o
[ 75%] Building C object CMakeFiles/zlibstatic.dir/trees.c.o
[ 83%] Building C object CMakeFiles/zlib.dir/uncompr.c.o
[ 83%] Building C object CMakeFiles/zlib.dir/trees.c.o
[ 83%] Building C object CMakeFiles/zlibstatic.dir/uncompr.c.o
[ 83%] Building C object CMakeFiles/zlib.dir/zutil.c.o
[ 83%] Building C object CMakeFiles/zlibstatic.dir/zutil.c.o
[ 86%] Linking C static library libz.a
[ 88%] Linking C static library libz.a
[ 88%] Built target zlib
[ 88%] Built target zlibstatic
Scanning dependencies of target minigzip
Scanning dependencies of target example
[ 94%] Building C object CMakeFiles/example.dir/test/example.c.o
[ 94%] Building C object CMakeFiles/minigzip.dir/test/minigzip.c.o
[100%] Linking C executable minigzip
[100%] Linking C executable example
[100%] Built target example
[100%] Built target minigzip
Install the project...
```

Even I downgraded version to v1.2.12:
```
Scanning dependencies of target zlibstatic
Scanning dependencies of target zlib
[  5%] Building C object CMakeFiles/zlibstatic.dir/adler32.o
[  5%] Building C object CMakeFiles/zlibstatic.dir/compress.o
[ 11%] Building C object CMakeFiles/zlibstatic.dir/crc32.o
[ 11%] Building C object CMakeFiles/zlibstatic.dir/deflate.o
[ 13%] Building C object CMakeFiles/zlibstatic.dir/gzclose.o
[ 16%] Building C object CMakeFiles/zlibstatic.dir/gzread.o
[ 22%] Building C object CMakeFiles/zlibstatic.dir/gzwrite.o
[ 22%] Building C object CMakeFiles/zlibstatic.dir/gzlib.o
[ 33%] Building C object CMakeFiles/zlib.dir/compress.o
[ 38%] Building C object CMakeFiles/zlib.dir/deflate.o
[ 44%] Building C object CMakeFiles/zlib.dir/adler32.o
[ 44%] Building C object CMakeFiles/zlib.dir/crc32.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzclose.o
[ 44%] Building C object CMakeFiles/zlibstatic.dir/inflate.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzlib.o
[ 44%] Building C object CMakeFiles/zlib.dir/gzread.o
[ 50%] Building C object CMakeFiles/zlib.dir/gzwrite.o
[ 50%] Building C object CMakeFiles/zlibstatic.dir/infback.o
[ 52%] Building C object CMakeFiles/zlib.dir/inflate.o
[ 61%] Building C object CMakeFiles/zlibstatic.dir/inftrees.o
[ 61%] Building C object CMakeFiles/zlibstatic.dir/inffast.o
[ 61%] Building C object CMakeFiles/zlib.dir/infback.o
[ 63%] Building C object CMakeFiles/zlibstatic.dir/trees.o
[ 66%] Building C object CMakeFiles/zlibstatic.dir/uncompr.o
[ 72%] Building C object CMakeFiles/zlibstatic.dir/zutil.o
[ 72%] Building C object CMakeFiles/zlib.dir/inftrees.o
[ 75%] Building C object CMakeFiles/zlib.dir/inffast.o
[ 77%] Building C object CMakeFiles/zlib.dir/trees.o
[ 80%] Building C object CMakeFiles/zlib.dir/uncompr.o
[ 83%] Building C object CMakeFiles/zlib.dir/zutil.o
[ 86%] Linking C static library libz.a
[ 86%] Built target zlibstatic
[ 88%] Linking C static library libz.a
[ 88%] Built target zlib
Scanning dependencies of target minigzip
Scanning dependencies of target example
[ 94%] Building C object CMakeFiles/example.dir/test/example.o
[ 94%] Building C object CMakeFiles/minigzip.dir/test/minigzip.o
[ 97%] Linking C executable minigzip
[100%] Linking C executable example
[100%] Built target example
[100%] Built target minigzip
Install the project...
```

I definitely saw some strange issues before but now it just somehow works for me. 😅 May be I can rebase this patch on top so it can be easly reverted later?

Reply:
In your log you can clearly see how it appears twice:
```
[ 86%] Linking C static library libz.a
[ 88%] Linking C static library libz.a
```
I’m pretty sure that if you compile with `make DEBUG=1` you will see that the link of the “libz.a“ apppears twice too

Reply:
Anyway… let’s merged with your changes, and we will see if this issue is no longer happening, if it fails we clearly know how to “fix it”